app/src/main/java/com/hapramp/datastore/DataStore.java

```diff
@@ -362,15 +362,27 @@ public void requestCommunityFeed(final String tag,
       public void run() {
         String url = URLS.curationUrl(tag);
         String cachedResponse = DataCache.get(url);
+        boolean cachedDataReturned = false;
         if (cachedResponse != null && isFeedRequestLive(rtag) && !refresh) {
           dispatchCommunityFeed(cachedResponse, false, false, userFeedCallback);
+          cachedDataReturned = true;
         }
         try {
           Response response = NetworkApi.getNetworkApiInstance().fetch(url);
           if (response.isSuccessful()) {
             String res = response.body().string();
             DataCache.cache(url, res);
-            if (isFeedRequestLive(rtag) && refresh) {
+            if (isFeedRequestLive(rtag)) {
+              if (!refresh && !cachedDataReturned) {
+                //no refresh requested but no cache found, so we de-respect the refresh-denial
+                // and return the fresh data.
+                dispatchCommunityFeed(res, true, false, userFeedCallback);
+                return;
+              }
+              if (refresh) {
+                //want to refresh
+                dispatchCommunityFeed(res, true, false, userFeedCallback);
```

Comment:
if `refresh` is true, both line 384 and 386 will run. Is that intended (they are same)?